torchx/lib/torchx.ex

```diff
@@ -1,6 +1,8 @@
 defmodule Torchx do
   alias Torchx.NIF
 
+  import Torchx.Backend, only: [torch_type: 1, torch_device: 1]
```

Comment:
We don't use torch_type here anymore, so we can remove it.

We also handle all of the device logic here, so perhaps move torch_device here too? 🤔